### ISSUE TITLE
SqlConnectionStringBuilder class missing HostNameInCertificate property

### ISSUE THREAD

**Issue by nemanja-milovancevic** [dotnet/SqlClient#1775]
### Describe the bug
SqlConnectionStringBuilder class is missing the HostNameInCertificate property on the net462 and net472. It's present on the net6.0.

### To reproduce

csproj file:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net462</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Data.SqlClient" Version="5.0.0" />
  </ItemGroup>

</Project>
```

C# file:
```c#
using Microsoft.Data.SqlClient;

namespace Example
{
    class Example
    {
        public static void Main()
        {
            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder();
            builder.HostNameInCertificate = "";
        }
    }
}
```

If you change the target version to net6.0 it will work. 

### Expected behavior
According to the [docs](https://github.com/dotnet/SqlClient/blob/main/release-notes/5.0/5.0.0.md) MDS5.0.0 should be supported on net462 and above. Important part of that is TDS 8.0 support.

### Further technical details
Microsoft.Data.SqlClient version: 5.0.0 
.NET target: Framework 4.6.2, 4.7.2
SQL Server version: N/A
Operating system: Windows Server 2019

**Additional context**
We're working on updating ADS to the TDS 8.0 and we can't do it without MDS support.

**Comment by lcheunglci** [comment]
Hi @nemanja-milovancevic , thanks for bringing the missing `HostNameInCertificate` property in the `SqlConnectionStringBuilder` to our attention. However, I did just quick check with your repro steps on .NET Framework 4.6.2 project, and verified that on `M.D.SqlClient 5.0 GA`, the HNIC is a public property for both .NET Core and .NET Framework and according to the implementation [here](https://github.com/dotnet/SqlClient/blob/04db1a40a95726af8567b7596653e20acdd7f19a/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionStringBuilder.cs#L1250), the changes applies to both netfx and netcore, so it should be supported.  There are also functional tests to check that HNIC is a valid keyword so it can be used when you specify it from connection string in the constructor of `SqlConnectionStringBuilder`.

**Comment by nemanja-milovancevic** [comment]
Interesting, it's working when I use `builder["HostNameInCertificate"] = ""` instead of `builder.HostNameInCertificate = ""`

**Comment by cheenamalhotra** [comment]
@lcheunglci 

The issue seems to be due to missing refs in .NET Framework ref project. Please find PR #1776 to fix the same.

**Comment by lcheunglci** [comment]
Thanks @cheenamalhotra !

**Comment by shueybubbles** [comment]
remind me - why do you bother maintaining separate "ref" DLLs instead of just having the compiler reference the full DLLs in the lib folder?

**Comment by lcheunglci** [comment]
@nemanja-milovancevic The fix has been merged in the 5.0.1 release.

**Comment by nemanja-milovancevic** [comment]
@cheenamalhotra @lcheunglci Thanks for fixing this!
